lib/rules/sort-keys.js

```diff
@@ -37,6 +37,38 @@ function getPropertyName(node) {
     return node.key.name || null;
 }
 
+/**
+ * Check the Line Separated Groups
+ *
+ * - Compare the start line of the current node with the last line of the previous node.
+ *   If the difference between the two values is greater than 0, Not the same group.
+ * @param {number} thisStartLine is start line of the current node.
+ * @param {number} prevEndLine is end line of the previous node.
+ * @param {Array<string>} sourceCodeLines use to check comment between groups.
+ * @returns {boolean} check group
+ * @private
+ */
+function checkLineSeparatedGroups(thisStartLine, prevEndLine, sourceCodeLines) {
+    let blockCommentFlag = false;
+    let commentOffset = 0;
+
+    sourceCodeLines.slice(prevEndLine, thisStartLine).forEach(value => {
+        if (value.includes("//")) {
+            commentOffset++;
+        } else if (value.includes("/*")) {
+            commentOffset++;
+            blockCommentFlag = true;
+        } else if (value.includes("*/")) {
+            commentOffset++;
+            blockCommentFlag = false;
+        } else if (blockCommentFlag) {
+            commentOffset++;
+        }
+    });
+
+    return !!Math.max(thisStartLine - prevEndLine - commentOffset - 1, 0);
+}
```

Comment:
It has some false positives when we use `/*`, `*/`, `//` as a value in code. 

```js
        {
            code: `
                var obj = {
                    c: "/*",

                    a: "*/", // Error. Expected object keys to be in ascending order. 'a' should be before 'c'.
                }
            `,
            options: ["asc", { allowLineSeparatedGroups: true }]
        }
```

> Is there a better way?

🤔 IMO, we can use [SourceCode methods](https://eslint.org/docs/developer-guide/working-with-rules#contextgetsourcecode) for it. there are some methods for treating comment as a `node or token`

```js
getTokenAfter(nodeOrToken, {includeComments: true})
getTokenBefore(nodeOrToken, {includeComments: true}); 
```

Reply:
Thank you @yeonjuan !!

in this case, 
```javascript
var sampleObject1 = {
  a: 1,
  b: 2,
  // comment 1
  // comment 2
  c: 3
}

I used `getTokensBefore`.

Reply:
I think this would be a complicated way to find out if there's a blank line.

For example, this is a false positive:

```js
/* eslint sort-keys: ["error", "asc", { allowLineSeparatedGroups: true }] */

var obj = {
  b,
  /*
  */ //

  a
}
```

comments offset is calculated as `3`, but it doesn't take into account if there are lines between comments or not.

Could we maybe simplify this? We could take tokens between properties and then see if there's a blank line between any two successive tokens.

More precisely, if we want to check if there's a blank line between two properties (`property1` and `property2`), than we could check if there's a blank line between any two of the following items:

* `property1 `
* comma after `property1`
* comment1
* comment2
* ....
* commentN
* `property2`